template.go

```diff
@@ -54,15 +64,51 @@ func readYamlRequest(opts *Options) error {
 		return err
 	}
 
-	headers, err := yaml.Marshal(t.Headers)
-	if err != nil {
-		return err
+	if t.Peer != "" {
+		opts.TOpts.HostPorts = []string{t.Peer}
+	} else if t.Peers != nil {
+		opts.TOpts.HostPorts = t.Peers
+	} else if t.PeerList != "" {
+		opts.TOpts.HostPortFile = path.Join(path.Dir(opts.ROpts.YamlTemplate), t.PeerList)
+	}
+
+	// Header arguments have precedence over template headers.
+	if t.Headers != nil {
```

Comment:
nit: prefer length check instead `len(t.Headers) > 0`
same below for `ROpts.Headers`